doc/source/contributing.rst

```diff
@@ -4,13 +4,574 @@
+dependencies.
+
+.. warning::
+
+   Sphinx version >= 1.2.2 or the older 1.1.3 is required.
+
+Building the documentation
+""""""""""""""""""""""""""
+
+So how do you build the docs? Navigate to your local  the folder
+``pandas/doc/`` directory in the console and run::
+
+    python make.py html
+
+And then you can find the html output in the folder ``pandas/doc/build/html/``.
+
+The first time it will take quite a while, because it has to run all the code
+examples in the documentation and build all generated docstring pages.
+In subsequent evocations, sphinx will try to only build the pages that have
+been modified.
+
+If you want to do a full clean build, do::
+
+    python make.py clean
+    python make.py build
+
+
+Starting with 0.13.1 you can tell ``make.py`` to compile only a single section
+of the docs, greatly reducing the turn-around time for checking your changes.
+You will be prompted to delete `.rst` files that aren't required, since the
+last committed version can always be restored from git.
+
+::
+
+    #omit autosummary and API section
+    python make.py clean
+    python make.py --no-api
+
+    # compile the docs with only a single
+    # section, that which is in indexing.rst
+    python make.py clean
+    python make.py --single indexing
+
+For comparison, a full documentation build may take 10 minutes. a ``-no-api`` build
+may take 3 minutes and a single section may take 15 seconds.  However, subsequent 
+builds only process portions you changed.  Now, open the following file in a web 
+browser to see the full documentation you just built::
+
+    pandas/docs/build/html/index.html
+
+And you'll have the satisfaction of seeing your new and improved documentation!
+
+
+Contributing to the code base
+-----------------------------
+
+.. contents:: Code Base:
+   :local:
+
+Code Standards
+^^^^^^^^^^^^^^
+
+*pandas* uses the `PEP8 <http://www.python.org/dev/peps/pep-0008/>`_ standard.  
+There are several tools to ensure you abide by this standard.
+
+We've written a tool to check that your commits are PEP8 great, `pip install pep8radius <https://github.com/hayd/pep8radius>`_. 
+Look at PEP8 fixes in your branch vs master with::
+
+    pep8radius master --diff` and make these changes with `pep8radius master --diff --in-place`
+
+Alternatively, use `flake8 <http://pypi.python.org/pypi/flake8>`_ tool for checking the style of your code.  
+Additional standards are outlined on the `code style wiki page <https://github.com/pydata/pandas/wiki/Code-Style-and-Conventions>`_.
+
+Please try to maintain backward-compatibility. *Pandas* has lots of users with lots of existing code, so 
+don't break it if at all possible.  If you think breakage is required clearly state why 
+as part of the Pull Request.  Also, be careful when changing method signatures and add 
+deprecation warnings where needed.
+
+Test-driven Development/Writing Code
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+*Pandas* is serious about `Test-driven Development (TDD) 
```

Comment:
I'd argue that a project can only be serious about *testing* but only an individual can be adhere to "TDD".
you can have the first without the second, and that's up to contributors. Is this just for "buzzword" points?